CRM/Member/Form/Membership.php

```diff
@@ -1849,7 +1849,8 @@ protected function getMembershipParameters(): array {
     }
 
     foreach ($this->order->getMembershipLineItems() as $membershipLineItem) {
-      $memTypeNumTerms = $this->getSubmittedValue('num_terms') ?: $membershipLineItem['membership_num_terms'];
+      $memTypeNumTerms = $this->getSubmittedValue('num_terms') &&
+      !$this->getSubmittedValue('price_set_id') ? $this->getSubmittedValue('num_terms') : $membershipLineItem['membership_num_terms'];
```

Comment:
Not clear why we would want to check if there is no price set id here. Wouldn't there always be in one if we're using a price set?
I think maybe the intent of what is supposed to be happening here is we are checking if there is a numeric value for a potential numeric price set field and then returning that or if there is no numeric value, then we're returning the number of terms from the price set option. But it's impossible to test that because numeric fields don't work for memberships in price sets at all. So for practical purposes, we could just get rid of this check and use $membershipLineItem['membership_num_terms'] directly. If we wanted to future proof this so it will work for future potential numeric price set fields, then I think we need to check the field type.
Either way, as things are currently set up $this->getSubmittedValue('num_terms') is always going to return 1, so we don't want that.